### PR TITLE
dataset creation rest api

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
This PR builds upon https://github.com/zowe/zss/pull/473 by trimming the list of parameters to those that are useful in dataset creation only.
Documentation of it is at https://github.com/zowe/zlux-server-framework/pull/441

## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: c/datasetjson.c [hunk 1]
```diff
@@ -2517,6 +2525,359 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
...
+          setRECFM = setRECFM | DALRECFM_V;
+        }
+        if (indexOf(valueString, valueStrLen, 'F', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_F;
+        }
+        if (indexOf(valueString, valueStrLen, 'U', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_U;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, setRECFM, DALRECFM, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "blkln")) {
```
Comment (1000TurquoisePogs) [inline c/datasetjson.c:2602]: this needs to be moved under "space" as an option. CYL, TRK, BLK???

File: c/datasetjson.c [hunk 2]
```diff
@@ -2517,6 +2525,359 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
...
+          }
+        }
+      } else if (!strcmp(propString, "ndisp")) {
+        if (!strcmp(valueString, "KEEP")){
+          parmDefn = DISP_KEEP;
+        } else {
+          parmDefn = DISP_CATLG;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, parmDefn, DALNDISP, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "unit")) {
```
Comment (1000TurquoisePogs) [inline c/datasetjson.c]: remove due to concern this could create hidden-to-zowe datasets

File: c/datasetjson.c [hunk 3]
```diff
@@ -2517,6 +2525,359 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
...
+  parmDefn = DISP_NEW; //ONLY one for create
+  rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, parmDefn, DALSTATS, configsCount, inputTextUnit);
+
+
+  // most parameters below explained here https://www.ibm.com/docs/en/zos/2.1.0?topic=dfsms-zos-using-data-sets
+  // or here https://www.ibm.com/docs/en/zos/2.1.0?topic=function-non-jcl-dynamic-allocation-functions
+  // or here https://www.ibm.com/docs/en/zos/2.1.0?topic=function-dsname-allocation-text-units
+  while(currentProp != NULL){
+    value = jsonPropertyGetValue(currentProp);
+    char *valueString = jsonAsString(value);
```
Comment (1000TurquoisePogs) [inline c/datasetjson.c]: everything as a string is silly, lets change this.

**Review by JoeNemo — APPROVED** [review]
Looks good

**Pull request merged**